examples/01_dirty_categories.py

```diff
@@ -1,398 +1,154 @@
-        (method, ["employee_position_title"]),
-        remainder="drop",
-    )
+dataset = fetch_employee_salaries()
 
-    pipeline = make_pipeline(encoder, HistGradientBoostingRegressor())
-    scores = cross_val_score(pipeline, X, y)
-    print(f"{name} encoding")
-    print(f"r2 score:  mean: {np.mean(scores):.3f}; std: {np.std(scores):.3f}\n")
-    all_scores[name] = scores
+dataset.description
 
 ###############################################################################
-# Plotting the results
-# ....................
-#
-# Finally, we plot the scores on a boxplot:
+# Alias *X*, the descriptions of employees (our input data), and *y*,
+# the annual salary (our target column):
 
-import matplotlib.pyplot as plt
-import seaborn
-
-plt.figure(figsize=(4, 3))
-ax = seaborn.boxplot(data=pd.DataFrame(all_scores), orient="h")
-plt.ylabel("Encoding", size=20)
-plt.xlabel("Prediction accuracy     ", size=20)
-plt.yticks(size=20)
-plt.tight_layout()
+X = dataset.X
+y = dataset.y
 
 ###############################################################################
-# The clear trend is that encoders grasping similarities between categories
-# (|SE|, |MinHash|, and |Gap|) perform better than those discarding it.
-#
-# |SE| is the best performer, but it is less scalable on big
-# data than the |MinHash| and |Gap|. The most scalable encoder is
-# the |MinHash|. On the other hand, the |Gap| has the benefit of
-# providing interpretable features
-# (see :ref:`sphx_glr_auto_examples_02_investigating_dirty_categories.py`)
-#
-# |
-#
+# And carry out some basic preprocessing
 
-###############################################################################
-# .. _example_table_vectorizer:
-#
-# A simpler way: automatic vectorization
-# --------------------------------------
-#
-# The code to assemble a column transformer is a bit tedious. We will
-# now explore a simpler, automated, way of encoding the data.
-#
-# Let's start again from the raw data:
-employee_salaries = fetch_employee_salaries()
-X = employee_salaries.X
-y = employee_salaries.y
+# Overload `employee_position_title` with 'underfilled_job_title',
+# as the latter gives more accurate job titles when specified
 
-###############################################################################
-# We'll drop the 'date_first_hired' column as it's redundant with
-# 'year_first_hired'.
-X = X.drop(["date_first_hired"], axis=1)
+X["employee_position_title"] = X["underfilled_job_title"].fillna(
+    X["employee_position_title"]
+)
+X.drop(labels=["underfilled_job_title"], axis="columns", inplace=True)
 
-###############################################################################
-# We still have a complex and heterogeneous dataframe:
 X
 
 ###############################################################################
-# The |TV| can to turn this dataframe into a form suited for
-# machine learning.
+# Let's extract a sample to have a more focused view: we'll limit our
```

Comment:
I would like that we first run the analysis (pipelining TableVectorizer with HBGT before discussing this

Reply:
Put it at the bottom, where we discuss the GapEncoder.